### PR TITLE
[HOTFIX] Drop duplicates in `Interactions`

### PR DESCRIPTION
**Pull request by nathancooperjones**
## Describe the big picture - why are we doing this?
If you had duplicate user/item pairs in an `Interactions` object before, they would still be represented twice in the sparse matrix in `Interactions`. This becomes a problem when we try to do evaluation, which requires we convert the COO matrix to a CSR matrix, and the following happens: 

Screen Shot 2021-04-27 at 10 20 48 AM

## Any additional details to clarify code in the PR?
I found the most efficient solution to this is to transfer the data over to a `dok_matrix`, then back to a `coo_matrix` during instantiation. I found that even on larger datasets, this adds only a few ms of overhead, and it's something we only need to do a single time. 

## How is this tested?
Added a unit test that failed in the `0.1.3` codebase but passes here. 

## Pull Request Checklist
 - [X] Pull request includes a description of why we are doing this
 - [X] ``CHANGELOG`` has been updated
 - [X] Version in ``_version.py`` has been updated
 - [X] All tests in the ``tests`` folder pass with a local build
 - [X] ``README`` has been updated (if applicable)
 - [ ] Documentation in ``docs`` has been updated (if applicable)
 - [ ] ``requirements.txt`` and ``requirements-dev.txt`` have been recompiled (if applicable)

## Screenshots or GIFs?
Screen Shot 2021-04-27 at 10 24 50 AM

### CODE REVIEW

File: tests/fixtures/interactions_fixtures.py [hunk 1]
```diff
@@ -80,6 +80,16 @@ def df_for_interactions_with_0_ratings():
     })
 
 
+@pytest.fixture()
+def df_for_interactions_with_duplicates():
+    # this should match ``df_for_interactions`` with duplicate user/item pairs added
+    return pd.DataFrame(data={
+        'user_id': [0, 0, 0, 1, 1, 2, 2, 2, 3, 3, 3, 4, 5, 5],
```
Comment (peterskipper) [inline tests/fixtures/interactions_fixtures.py:87]: I don't see any duplicates here? Should a dupe be identical user/item/rating triplets, or identical user/item pairs? I don't see either.

**Review by peterskipper — CHANGES_REQUESTED** [review]
Good catch! I left a question on the new test because I 'm not sure how it works?

File: tests/fixtures/interactions_fixtures.py (hunk 1 above)
Comment (nathancooperjones) [reply tests/fixtures/interactions_fixtures.py:87]: Duplicate user/item pairs are at indices `0 & 1` and `12 & 13`. I'll leave a comment with this so it's clearer.

Comment (peterskipper) [reply tests/fixtures/interactions_fixtures.py:88]: ugh yup I see it. Sorry!

**Review by peterskipper — APPROVED** [review]
LGTM

**Pull request merged**